versioning/SemverVersioning.test.js

```diff
@@ -0,0 +1,191 @@
+                    '1.2.0': { enabled: true, mandatory: false, ...MOCK_INFOS },
+                };
+
+                expect(new SemverVersioning(RELEASED_BUNDLES).checkIsMandatory('1.0.0')).toBe(true);
+                expect(new SemverVersioning(RELEASED_BUNDLES).checkIsMandatory('1.0.1')).toBe(false);
+                expect(new SemverVersioning(RELEASED_BUNDLES).checkIsMandatory('1.1.0')).toBe(false);
+                expect(new SemverVersioning(RELEASED_BUNDLES).checkIsMandatory('1.2.0')).toBe(false);
+            });
+        
+            it('Major Release > Mandatory > Not-mandatory > Mandatory > Not-mandatory', () => {
+                const RELEASED_BUNDLES = {
+                    '1.0.0': FIRST_RELEASE_INFO,
+                    '1.0.1': { enabled: true, mandatory: true, ...MOCK_INFOS },
+                    '1.1.0': { enabled: true, mandatory: false, ...MOCK_INFOS },
+                    '1.1.1': { enabled: true, mandatory: true, ...MOCK_INFOS },
+                    '1.2.0': { enabled: true, mandatory: false, ...MOCK_INFOS },
+                };
+
+                expect(new SemverVersioning(RELEASED_BUNDLES).checkIsMandatory('1.0.0')).toBe(true);
+                expect(new SemverVersioning(RELEASED_BUNDLES).checkIsMandatory('1.0.1')).toBe(true);
+                expect(new SemverVersioning(RELEASED_BUNDLES).checkIsMandatory('1.1.0')).toBe(true);
+                expect(new SemverVersioning(RELEASED_BUNDLES).checkIsMandatory('1.1.1')).toBe(false);
+                expect(new SemverVersioning(RELEASED_BUNDLES).checkIsMandatory('1.2.0')).toBe(false);
+            });
+        })
+    })
+
+    describe('shouldRollback', () => {
+        it('should return true when latest version < current runtime version', () => {
+            const RELEASED_BUNDLES = {
+                '1.1.0': { enabled: true, mandatory: false, ...MOCK_INFOS },
+            };
+            const currentVersion = '1.2.0'
+
+            expect(new SemverVersioning(RELEASED_BUNDLES).shouldRollback(currentVersion)).toBe(true)
+        })
+    })
+
+    describe('shouldRollbackToLatestMajorVersion', () => {
+        it('should return false if it is not required to rollback', () => {
+            const RELEASED_BUNDLES_1 = {
+                '1.0.0': { enabled: true, mandatory: false, ...MOCK_INFOS },
+            };
+            const RELEASED_BUNDLES_2 = {
+                '1.0.0': { enabled: true, mandatory: false, ...MOCK_INFOS },
+                '1.1.0': { enabled: true, mandatory: false, ...MOCK_INFOS },
+                '1.2.0': { enabled: true, mandatory: false, ...MOCK_INFOS },
+            };
+            expect(new SemverVersioning(RELEASED_BUNDLES_1).shouldRollbackToLatestMajorVersion('1.0.0')).toBe(false)
+            expect(new SemverVersioning(RELEASED_BUNDLES_2).shouldRollbackToLatestMajorVersion('1.1.0')).toBe(false)
+        })
+
+        it('should return true if the rollback version is the latest major version', () => {
```

Comment:
Does "the rollback version" mean the destination version?
It would be better to clarify this.